assets/src/edit-story/components/IconSwitch/index.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * External dependencies
+ */
+import styled from 'styled-components';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import LockClosedIcon from './lock_closed.svg';
+import LockOpenIcon from './lock_open.svg';
+
+const StyledButton = styled.button`
+  border: none;
+  background: transparent;
+  display: block;
+	padding: 0;
+	width: ${ ( { width } ) => width }px;
+	height: ${ ( { height } ) => height }px;
+  min-width: initial;
+  cursor: pointer;
+	color: ${ ( { theme } ) => theme.colors.fg.v3 };
+	&:focus, &:active, &:hover {
+    opacity: 1;
+    outline: none;
+	}
+	svg {
+		width: ${ ( { width } ) => width }px;
+		height: ${ ( { height } ) => height }px;
+  }
+  ${ ( { disabled } ) => disabled && `
+		pointer-events: none;
+		opacity: .3;
+	` }
+`;
+
+const IconSwitch = ( { locked, ...props } ) => (
+	<StyledButton { ...props }>
+		{ locked ? <LockClosedIcon /> : <LockOpenIcon /> }
+	</StyledButton>
+);
```

Comment:
This ought to be an invisible `<input type="checkbox" />` surrounded by a label and followed by a `<span />` showing the state of the toggle. Basically google "custom checkbox in css". This will make it a lot more accessible and semantic.

Reply:
That makes sense for me. Currently I implemented with general approach with label, input, span.
And uses Icon directly inside span element. Actually I was trying to use icons as background-image of span, but no idea how to do it. What's your opinion of this two different approaches? Maybe in performance perspective.

Reply:
If you had full access to the design, you would be able to find the generic checkbox icons there. For now, I think this is fine 👍